internal/definition/awsintegration/schema_integration.go

```diff
@@ -319,3 +320,90 @@ func decodeTerraform(rd *schema.ResourceData) (*integration.AwsCloudWatchIntegra
 
 	return cwi, nil
 }
+
+func encodeTerraform(aws *integration.AwsCloudWatchIntegration, d *schema.ResourceData) error {
+	errs := multierr.Combine(
+		d.Set("integration_id", aws.Id),
+		d.Set("name", aws.Name),
+		d.Set("enabled", aws.Enabled),
+		d.Set("enable_aws_usage", aws.EnableAwsUsage),
+		d.Set("import_cloud_watch", aws.ImportCloudWatch),
+		d.Set("poll_rate", aws.PollRate/1000),
+		d.Set("use_metric_streams_sync", aws.MetricStreamsSyncState == "ENABLED"),
+		d.Set("enable_logs_sync", aws.LogsSyncState == "ENABLED"),
+		d.Set("enable_check_large_volume", aws.EnableCheckLargeVolume),
+		d.Set("sync_custom_namespaces_only", aws.SyncCustomNamespacesOnly),
+		d.Set("named_token", aws.NamedToken),
+		d.Set("collect_only_recommended_stats", aws.CollectOnlyRecommendedStats),
+		d.Set("metric_streams_managed_externally", aws.MetricStreamsManagedExternally),
+	)
+
+	if aws.Token != "" {
```

Comment:
Hmm, I thought I commented on this last week but it looks like I didn't...

Anyway, shouldn't we check for `aws.Key` too?

Reply:
Fixed in https://github.com/splunk-terraform/terraform-provider-signalfx/pull/580

Reply:
Added :D